specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json

```diff
@@ -117,6 +117,127 @@
         }
       }
     },
+    "/{scope}/providers/Microsoft.Consumption/reservationsummaries": {
+      "get": {
+        "tags": [
+          "ReservedInstances"
+        ],
+        "operationId": "ReservationsSummaries_List",
+        "description": "Lists the reservations summaries for daily or monthly grain.",
+        "externalDocs": {
+          "url": "https://azure.microsoft.com/en-us/blog/announcing-preview-of-consumption-and-charge-apis-for-enterprise-azure-customers/"
+        },
+        "x-ms-examples": {         
+          "ReservationSummariesDaily": {
+            "$ref": "./examples/ReservationSummariesDaily.json"
+          },
+          "ReservationSummariesDailyWithReservationId": {
+            "$ref": "./examples/ReservationSummariesDailyWithReservationId.json"
+          },
+          "ReservationSummariesMonthly": {
+            "$ref": "./examples/ReservationSummariesMonthly.json"
+          },
+          "ReservationSummariesMonthlyWithReservationId": {
+            "$ref": "./examples/ReservationSummariesMonthlyWithReservationId.json"
+          }
+          },
+        "parameters": [
+          {
+            "name": "scope",
+            "description": "The scope of the reservation summaries. The scope can be 'providers/Microsoft.Capacity/reservationorders/{ReservationOrderId}' or 'providers/Microsoft.Capacity/reservationorders/{ReservationOrderId}/reservations/{ReservationId}'",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "x-ms-skip-url-encoding": true
+          },  
+          {
+            "name": "grain",
+            "description": "Can be daily or monthly",
+            "in": "query",
+            "required": true,
+            "type": "string"
+          },       
+          {
+            "name": "$filter",
+            "description": "Required only for daily grain. The properties/UsageDate for start date and end date. The filter supports 'le' and  'ge'",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ReservationSummariesListResult"
```

Comment:
would you want to add x-ms-pageable to this operation- https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-pageable 
You can set nextLinkName to null, which results in an enumerable for some of the SDKs.

Reply:
pushed my changes